### PR TITLE
[Merged by Bors] - Instrument SPU

### PR DESCRIPTION
**Pull request by nicholastmosher**
- Adds `#[tracing::instrument]` to more areas in the SPU to support better tracing e.g. with otel/jaeger

### CODE REVIEW

**Comment by sehz** [comment]
rebase

**Comment by sehz** [comment]
rebase to make CI success

File: src/spu/src/control_plane/message_sink.rs [hunk 1]
```diff
@@ -12,17 +13,20 @@ pub type SharedStatusUpdate = Arc<StatusMessageSink>;
 pub struct StatusMessageSink(Mutex<HashSet<LrsRequest>>);
 
 impl StatusMessageSink {
+    #[instrument]
```
Comment (sehz) [inline src/spu/src/control_plane/message_sink.rs:16]: too low level. remove this

File: src/spu/src/control_plane/message_sink.rs [hunk 2]
```diff
@@ -12,17 +13,20 @@ pub type SharedStatusUpdate = Arc<StatusMessageSink>;
...
     pub fn shared() -> Arc<Self> {
         Arc::new(Self(Mutex::new(HashSet::new())))
     }
 
     /// send lrs request sc
     /// newer entry will overwrite previous if it has not been cleared
+    #[instrument(skip(self, request))]
```
Comment (sehz) [inline src/spu/src/control_plane/message_sink.rs:23]: remove

File: src/spu/src/control_plane/message_sink.rs [hunk 3]
```diff
@@ -12,17 +13,20 @@ pub type SharedStatusUpdate = Arc<StatusMessageSink>;
...
     pub async fn send(&self, request: LrsRequest) {
         let mut lock = self.0.lock().await;
         lock.replace(request);
     }
 
+    #[instrument(skip(self))]
```
Comment (sehz) [inline src/spu/src/control_plane/message_sink.rs:29]: remove

File: src/spu/src/replication/leader/replica_state.rs [hunk 4]
```diff
@@ -274,6 +274,7 @@ where
     }
 
     /// convert myself as
+    #[instrument(skip(self))]
```
Comment (sehz) [inline src/spu/src/replication/leader/replica_state.rs:277]: remove

**Review by sehz — CHANGES_REQUESTED** [review]
just couple of nits

**Comment by nicholastmosher** [comment]
@sehz rebased

**Review by sehz — APPROVED** [review]
LGTM

**Comment by nicholastmosher** [comment]
bors ping

**Pull request closed without merging**